In [5]:
import mindspore
from loss import *
from sklearn.metrics import matthews_corrcoef, confusion_matrix

In [6]:
gt_mask = mindspore.Tensor([
    [0, 0, 0, 0, 0],
    [0, 0, 1, 0, 0],
    [0, 1, 1, 1, 0],
    [0, 1, 1, 1, 1],
    [0, 0, 1, 1, 0]
])

pred_mask = mindspore.Tensor([
    [0, 0, 0, 0, 0],
    [0, 1, 1, 0, 0],
    [0, 1, 1, 1, 0],
    [0, 1, 1, 1, 0],
    [0, 0, 0, 0, 0]
])

In [7]:
dice_loss = Dice_Loss()
mcc_loss = MCC_Loss()

In [12]:
# Dice coefficient = 1 - Dice loss (since Dice loss = 1 - Dice coefficient in loss.py)
1 - dice_loss(pred_mask, gt_mask).item()

0.7777777910232544

In [13]:
# MCC = 1 - MCC loss (since MCC loss = 1 - MCC in loss.py)
1 - mcc_loss(pred_mask, gt_mask).item()

0.6605085730552673

## Checking the correctness of the loss functions

In [10]:
# Confusion matrix created using scikit-learn
confusion_matrix(y_true=gt_mask.flatten().numpy(), y_pred=pred_mask.flatten().numpy())

array([[14,  1],
       [ 3,  7]])

In [11]:
# Matthews correlation coefficient calculated using scikit-learn
matthews_corrcoef(y_true=gt_mask.flatten().numpy(), y_pred=pred_mask.flatten().numpy())

0.6651330399133047

### Dice coefficient = 2 * (intersection of gt_mask and pred_mask) / (|gt_mask| + |pred_mask|) = (2*14) / (10 + 8) = 0.7778

### The difference in MCC values are because of the (+ 1) in the denominator, which is added to avoid divide-by-zero errors.